lib/libc/stdlib/strfmon.c

```diff
@@ -128,7 +128,6 @@ vstrfmon_l(char * __restrict s, size_t maxsize, locale_t loc,
 
 	char		*tmpptr;	/* temporary vars */
 	int		sverrno;
-	FIX_LOCALE(loc);
```

Comment:
Can you explain the move of FIX_LOCALE() call from there to strfmon_l()? In other words, why does strfmon() not need the fixup?

Reply:
Maybe I misinterpreted:

> Any public function that accepts a `locale_t` parameter must use the `FIX_LOCALE()` macro on it before using it.  For efficiency, this can be emitted (sic) in functions which *only* use their locale parameter as an argument to another public function, as the callee will do the `FIX_LOCALE()` itself.

And moved it to:

https://github.com/freebsd/freebsd-src/blob/70059838ae4ca5621c4c3c5cdf978a0db5bf838e/lib/libc/stdlib/strfmon.c#L662

I'll put it back to where it was before.

Reply:
Yes, but you left strfmon() without FIX_LOCALE(), this is what I am asking about.

Even if the move is legitimate (and I misunderstand it), it should be a separate change with appropriate commit message explaining why.

Reply:
The original reasoning was:
1. `strfmon` uses `__get_locale()` to supply the `locale_t` parameter, no need to `FIX_LOCALE()`.
2. `strfmon_l` do need `FIX_LOCALE()`. So, move the `FIX_LOCALE()` to the corresponding function.

Anyhow, it's reverted back to `vstrfmon_l`, where it was originally.

Reply:
Ok, please provide this move as a separate change then.  You might just mail me the git-format patch after this series goes in.